app/services/facets.rb

```diff
@@ -0,0 +1,24 @@
+module Facets
+  module_function
+
+  def registered_facets
+    configuration.dup
+  end
+
+  def register(facet_model, facet_name = nil, &block)
+    entry = Facets::Entry.new(facet_model, facet_name)
+
+    entry.instance_eval(&block) if block_given?
```

Comment:
You can make this slightly faster by not passing `&block` as an argument, check out http://mudge.name/2011/01/26/passing-blocks-in-ruby-without-block.html . I don't think it matters a lot though, but a comment above the method giving an example of facet that would use `block` would help. I can't remember any right now.